packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,262 @@
+           alignment == InlineWidgetAlignment.belowBaseline ? ui.PlaceholderAlignment.belowBaseline :
+           alignment == InlineWidgetAlignment.top ? ui.PlaceholderAlignment.top :
+           alignment == InlineWidgetAlignment.bottom ? ui.PlaceholderAlignment.bottom :
+           alignment == InlineWidgetAlignment.middle ? ui.PlaceholderAlignment.middle : null,
+         baseline: baseline, style: style, children: children,
+       );
+
+  /// The widget to embed inline with text.
+  final Widget widget;
+
+  /// Adds a placeholder box to the paragraph builder if a size has been
+  /// calculated for the widget.
+  ///
+  /// Sizes are provided through [dimensions], which should contain a 1:1
+  /// in-order mapping of widget to laid out dimensions. If no such dimension
+  /// is provided, the widget will be skipped.
+  ///
+  /// Since widget sizes are calculated independently from the rest of the
+  /// paragraph, the [textScaleFactor] is ignored.
+  @override
+  void build(ui.ParagraphBuilder builder, { double textScaleFactor = 1.0, List<PlaceholderDimensions> dimensions }) {
+    assert(debugAssertIsValid());
+
+    final bool hasStyle = style != null;
+    if (hasStyle)
+      builder.pushStyle(style.getTextStyle(textScaleFactor: textScaleFactor));
+    if (dimensions != null && builder.placeholderCount < dimensions.length) {
+      PlaceholderDimensions currentDimensions = dimensions[builder.placeholderCount];
+      builder.addPlaceholder(
+        currentDimensions.size.width,
+        currentDimensions.size.height,
+        alignment,
+        baseline: currentDimensions.baseline,
+        baselineOffset: currentDimensions.baselineOffset,
+      );
+    }
+    if (hasStyle)
+      builder.pop();
+  }
+
+  /// Calls visitor on this [WidgetSpan]. There are no children spans to walk.
+  bool visitChildren(bool visitor(InlineSpan span)) {
+    if (!visitor(this))
+      return false;
+    if (children != null) {
+      for (InlineSpan child in children) {
+        if (!child.visitChildren(visitor))
+          return false;
+      }
+    }
+    return true;
+  }
+
+  String toPlainText() {
+    return widget.toString();
+  }
+
+  int codeUnitAt(int index) {
+    return null;
+  }
+
+  /// Describe the difference between this widget span and another [TextSpan],
+  /// in terms of how much damage it will make to the rendering. The comparison
+  /// is deep.
+  ///
+  /// Comparing a [WidgetSpan] with a [TextSpan] will result in [RenderComparison.layout].
+  ///
+  /// See also:
+  ///
+  ///  * [TextStyle.compareTo], which does the same thing for [TextStyle]s.
+  RenderComparison compareTo(InlineSpan other) {
+    if (identical(this, other))
+      return RenderComparison.identical;
+    if (other.runtimeType is! WidgetSpan)
+      return RenderComparison.layout;
+    final WidgetSpan typedOther = other;
+    if (typedOther.widget != widget)
+      return RenderComparison.layout;
```

Comment:
i'm not sure that's necessary. If the widgets don't match, the presumably you'll find out about it when the widgets themselves are updated.